entity-framework/core/querying/complex-query-operators.md

```diff
@@ -0,0 +1,128 @@
+
+SELECT [b].[BlogId], [b].[OwnerId], [b].[Rating], [b].[Url], [p].[PostId], [p].[AuthorId], [p].[BlogId], [p].[Content], [p].[Rating], [p].[Title]
+FROM [Blogs] AS [b]
+LEFT JOIN [Posts] AS [p] ON [b].[BlogId] = [p].[BlogId]
+```
+
+### Collection selector references outer in non-where case
+
+When the collection selector has reference to the outer element, which isn't in where clause as the case above, it doesn't translate to database join. That's why we need to evaluate the collection selector for each outer element. It translates to `APPLY` operation in many relational databases. If the collection is empty for outer element, then no results would be generated for that outer element. But if `DefaultIfEmpty` is applied on the collection selector then the outer element will be connected with default value of the inner element. Because of this distinction, this kind of queries translates to `CROSS APPLY` in the absence of `DefaultIfEmpty` and `OUTER APPLY` when `DefaultIfEmpty` is applied. Certain databases like SQLite don't support `APPLY` operator so this kind of query may not be translated.
+
+[!code-csharp[Main](../../../samples/core/Querying/ComplexQuery/Sample.cs#SelectManyConvertedToApply)]
+
+```SQL
+SELECT [b].[BlogId], [b].[OwnerId], [b].[Rating], [b].[Url], ([b].[Url] + N'=>') + [p].[Title] AS [p]
+FROM [Blogs] AS [b]
+CROSS APPLY [Posts] AS [p]
+
+SELECT [b].[BlogId], [b].[OwnerId], [b].[Rating], [b].[Url], ([b].[Url] + N'=>') + [p].[Title] AS [p]
+FROM [Blogs] AS [b]
+OUTER APPLY [Posts] AS [p]
+```
+
+## GroupBy
+
+LINQ GroupBy operator creates result of type `IGrouping<TKey, TElement>` where `TKey` and `TElement` could be arbitrary types. Further, `IGrouping` implements `IEnumerable<TElement>`, which means you can compose over using any LINQ operator after grouping. Since no database structure can represent `IGrouping`, Group By operator has no translation in most cases. When an aggregate operator is applied to each group, which gives back a scalar, it can be translated to SQL `GROUP BY` in relational databases. SQL `GROUP BY` is restrictive too. It requires you to group by only scalar values  projection can contain only grouping key columns or aggregate applied over any column. EF Core identifies this subset and translates it to server, as in the following example:
+
+[!code-csharp[Main](../../../samples/core/Querying/ComplexQuery/Sample.cs#GroupBy)]
+
+```SQL
+SELECT [p].[AuthorId] AS [Key], COUNT(*) AS [Count]
+FROM [Posts] AS [p]
+GROUP BY [p].[AuthorId]
+```
+
+EF Core also translates queries where an aggregate operator on the grouping appears in Where or OrderBy (or other ordering) LINQ operators. It uses `HAVING` clause in SQL for where clause. The query part before applying GroupBy operator can be any complex query as long as it can be translated to server. Further, once you apply aggregate operators on a grouping query to remove groupings from resulting source, you can compose it like any other query.
```

Comment:
```suggestion
EF Core also translates queries where an aggregate operator on the grouping appears in a Where or OrderBy (or other ordering) LINQ operator. It uses `HAVING` clauses in SQL for the where clause. The part of the query before applying the GroupBy operator can be any complex query as long as it can be translated to server. Furthermore, once you apply aggregate operators on a grouping query to remove groupings from the resulting source, you can compose on top of it like any other query.
```